tripy/tripy/frontend/module/sequential.py

```diff
@@ -0,0 +1,186 @@
+        else:
+            for idx, module in enumerate(modules):
+                self.modules[str(idx)] = module
+
+    def __call__(self, input: "tripy.Tensor") -> "tripy.Tensor":
+        r"""
+        Defines the forward pass by applying each module in the container sequentially to `input`
+
+        Args:
+            input: The input tensor to pass through the sequence of modules.
+
+        Returns:
+            The output tensor after passing through each module in sequence.
+        """
+        for module in self.modules.values():
+            input = module(input)
+        return input
+
+    def __getattr__(self, name) -> Any:
+        """
+        Custom __getattr__ to search both in `modules` dictionary and in other attributes. This is for handling
+        `module = operator.attrgetter(child_name)(module)` calls in tripy/frontend/module/module.py:load_state_dict
+        """
+        # Check if `name` is a key in the modules dictionary
+        if "modules" in self.__dict__ and name in self.modules:
+            return self.modules[name]
+
+        # Fallback to regular attribute access if not found in modules
+        return super().__getattr__(name)
+
+    def __len__(self) -> int:
+        r"""
+        Returns the total number of modules in the sequence.
+
+        Returns:
+            The number of modules in the sequence.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            # doc: print-locals model length
+
+            model = tp.Sequential(tp.Linear(1, 64), tp.Linear(64, 128))
+            length = len(model)
+            assert length == 2
+        """
+        return len(self.modules)
+
+    def __iter__(self) -> Iterator[Module]:
+        r"""
+        Returns an iterator over the modules in the sequence.
+
+        Returns:
+            An iterator over the modules.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+            for layer in model:
+                print(layer)
+        """
+        return iter(self.modules.values())
+
+    def __getitem__(self, idx: Union[int, str]) -> Module:
+        r"""
+        Accesses a module by index (int) or name (str).
+
+        Args:
+            idx: The index or name of the module to retrieve.
+
+        Returns:
+            The module at the specified index or name.
+
+        Raises:
+            TypeError: If `idx` is not an int or str.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+            print(model[1])
+        """
+        if not isinstance(idx, (int, str)):
+            raise TypeError("Index must be an int or str.")
+
+        key = str(idx) if isinstance(idx, int) else idx
+
+        if key not in self.modules:
+            raise ValueError(f"Key {key} not found in modules.")
```

Comment:
Let's use `raise_error`:
```suggestion
            raise_error(f"Key: '{key}' not found in modules.", [f"Note: Available keys were: {list(self.modules.keys())}"])
```